tests/test_data.py

```diff
@@ -0,0 +1,187 @@
+from json import dumps
+import pytest
+from time import sleep
+
+from data import DataFactory, DataV2, DataV3
+import topics
+
+def test_data_factory():
+    assert isinstance(DataFactory.create("v2"), DataV2)
+    assert isinstance(DataFactory.create("V3"), DataV3)
+    with pytest.raises(NotImplementedError):
```

Comment:
`test_data_factory` should be two tests that are split between a successful path and an error path.